### PR TITLE
Adding Useful Django Modules

### PR DESCRIPTION
**Pull request by ShyamSundhar1411**
Added handy Django modules which are not known to many users but which when learnt might make their works easy

## What does this PR do?
I have added Django modules which are not known to many people but when put into exercise can make their works easy.

### Description
Added handy Django modules which are not known to many users but which when learnt might make their works easy such as Django Filters, Django-Storages, Django-Allauth, Django Autocomplete-light

### How do we know it's really free?
It's from open-source platforms like readthedocs.io and open-source GitHub repositories
## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by eshellman** [comment]
We proably should link to the web versions of readthedocs documentation since they have links to the [df (and often EPUB) versions.

**Comment by ShyamSundhar1411** [comment]
I will look on to it

**Comment by ShyamSundhar1411** [comment]
I have made the requested Changes. Kindly review it..

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -2002,12 +2005,16 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [Django Official Documentation](https://buildmedia.readthedocs.org/media/pdf/django/2.2.x/django.pdf) (PDF) (2.2)
 * [Django Official Documentation](https://buildmedia.readthedocs.org/media/pdf/django/3.1.x/django.pdf) (PDF) (3.1)
 * [Django Official Documentation](https://buildmedia.readthedocs.org/media/pdf/django/3.2.x/django.pdf) (PDF) (3.2)
+* [Django Official Documentation](https://buildmedia.readthedocs.org/media/pdf/django/4.0.x/django.pdf) (PDF) (4.0)
 * [Django Rest Framework](https://riptutorial.com/Download/django-rest-framework.pdf) - Compiled from StackOverflow Documentation (PDF)
 * [Django RESTful Web Services](https://www.packtpub.com/free-ebooks/django-restful-web-services) - Gaston C. Hillar (Packt account *required*)
+* [Django Storages](https://django-storages.readthedocs.io/en/latest/)(HTML)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:2011]:
```suggestion
* [Django Storages](https://django-storages.readthedocs.io/en/latest/) (HTML)
```

File: books/free-programming-books-langs.md [hunk 2]
```diff
@@ -2002,12 +2005,16 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
...
+* [Django Tinymce](https://django-tinymce.readthedocs.io/en/latest/)(HTML)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:2012]:
```suggestion
* [Django Tinymce](https://django-tinymce.readthedocs.io/en/latest/) (HTML)
```

File: books/free-programming-books-langs.md [hunk 3]
```diff
@@ -2002,12 +2005,16 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
...
 * [Django Web Framework (Python)](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Django) - MDN contributors
 * [Djen of Django](http://agiliq.com/books/djenofdjango/)
 * [Effective Django](https://web.archive.org/web/20181130092020/http://www.effectivedjango.com/) (1.5)
 * [How to Tango With Django](http://www.tangowithdjango.com/book17/) (1.7)
+* [Social Auth App](https://python-social-auth.readthedocs.io/en/latest/)(HTML)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:2017]:
```suggestion
* [Social Auth App](https://python-social-auth.readthedocs.io/en/latest/) (HTML)
```

File: books/free-programming-books-langs.md [hunk 4]
```diff
@@ -1994,6 +1994,9 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 
 #### Django
 
+* [All-Auth](https://django-allauth.readthedocs.io/en/latest/overview.html)(HTML)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:1997]:
```suggestion
* [All-Auth](https://django-allauth.readthedocs.io/en/latest/overview.html) (HTML)
```

File: books/free-programming-books-langs.md [hunk 5]
```diff
@@ -1994,6 +1994,9 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
...
+* [AutoComplete-Light](https://django-autocomplete-light.readthedocs.io/en/master/tutorial.html)(HTML)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:1998]:
```suggestion
* [AutoComplete-Light](https://django-autocomplete-light.readthedocs.io/en/master/tutorial.html) (HTML)
```

File: books/free-programming-books-langs.md [hunk 6]
```diff
@@ -1994,6 +1994,9 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
...
+* [Django Filters](https://django-filter.readthedocs.io/en/stable/)(HTML)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:1999]:
```suggestion
* [Django Filters](https://django-filter.readthedocs.io/en/stable/) (HTML)
```

**Review by eshellman — CHANGES_REQUESTED** [review]

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**